docs/stories/WebTools.md

```diff
@@ -0,0 +1,53 @@
+# Web Tools
+
+## Configure web application directory in WTP configurations for web application projects
+
+### Motivation
+The `ide-integration.md` document in Gradle core specifies a story ["Expose web application directory"](...). The goal is to define an initial end-to-end framework for Buidship/WTP integration, and to
+communicate the location of the web application directory to WTP.
+
+### Estimate
+
+- ? day(s)
+
+### Implementation
+- Create branch in buildship repository `wtp-integration`.
+- Define `org.eclipse.buildship.wtp` plugin in `wtp-integration`.
+- Define `org.eclipse.buildship.wtp.test` plugin in `wtp-integration`.
+- Wait for corresponding story in TAPI to be completed.
+- Upgrade Tooling Commons (if necessary).
+- Define configuration class in `org.eclipse.buildship.wtp`, either configurator class, or a similar class to `DefaultWorkspaceGradleOperations`.
+    - Retrieve Web application model from TAPI
+    - If a project is a web application project (model will be `null` if project doesn't apply `war` plugin)
+        - Add Java and Dynamic Web Facets to project, which will generate component file
+        - Configure web app dir location in component file
+    - Otherwise, do nothing
+- Invoke WTP configuration from Buildship core
+
+### Test cases
+- Configuration only invoked on web application projects
+- Java facet is added to project
+- Dynamic Web Facet is added to project
+- If Java Facet already on project but has incorrect version, version is updated to correct version based on source Java version
+- If Dynamic Web Facet already on project but has incorrect version, version is updated to correct version
+- Default Java Facet version is x.
+- Default Dynamic Web Facet version is x.
+- If Java Facet exists on project with correct version, Facet is not affected.
+- If Dynamic Web Facet exists on project with correct version, Facet is not affected.
+- Custom web application directory location is communicated successfully to component file, file exists.
+- Default web application directory location is communicated successfully to component file, file exists.
+- Custom web application directory location is communicated successfuly to component file, file does not exist.
+- Default web application directory location is communicated successfully to component file, file does not exist.
+
```

Comment:
My questions may have been a bit convoluted.

> you do want to see /src/main/webapp changed into /src/main/ian if the build.gradle file changed its webappdirname. build.gradle is the truth. tag=defaultRootSource should always match gradle truth. Even if no entries existed before in .component.
But if other entries exist in that file, I think they should remain there.

Okay, I'll create a test accordingly. My example was an example of a different test, but was meant to show that such a test cannot test whether the WTP plugin persists entries in the component file.

If the build.gradle file does not define a webAppDir location, but the component file of an imported project does, is the web app dir entry in the component file overwritten with the default webAppDir location taken from the Gradle model?

What should happen if a project is imported that doesn't have a web app dir defined in the component file?